sdk/java/pulumi/src/test/java/com/pulumi/automation/LocalWorkspaceTest.java

```diff
@@ -39,15 +56,447 @@ private static String randomStackName() {
+                .environmentVariables(env)
+                .build())) {
+
+            var stackName = randomStackName();
+
+            Predicate<StackSummary> isStack = s -> s.name().equals(stackName);
+
+            var stacks = workspace.listStacks();
+            if (stacks.stream().anyMatch(isStack)) {
+                workspace.removeStack(stackName);
+                stacks = workspace.listStacks();
+                assertThat(stacks).noneMatch(isStack);
+            }
+
+            workspace.createStack(stackName);
+            stacks = workspace.listStacks();
+            var newStack = stacks.stream().filter(isStack).findFirst();
+            assertThat(newStack).isPresent().hasValueSatisfying(s -> {
+                assertThat(s.isCurrent()).isTrue();
+            });
+
+            workspace.selectStack(stackName);
+            workspace.removeStack(stackName);
+            stacks = workspace.listStacks();
+            assertThat(stacks).noneMatch(isStack);
+        }
+    }
+
+    @Test
+    @Timeout(value = 1, unit = TimeUnit.MINUTES)
+    void testImportExportStack(@EnvVars Map<String, String> envVars) throws Exception {
+        var env = new HashMap<String, String>(envVars);
+        env.put("PULUMI_CONFIG_PASSPHRASE", "test");
+
+        Path workingDir = Paths.get(getClass().getResource("/testproj").toURI());
+        var projectSettings = ProjectSettings.builder("testproj", ProjectRuntimeName.GO)
+                .description("A minimal Go Pulumi program")
+                .build();
+
+        try (var workspace = LocalWorkspace.create(LocalWorkspaceOptions.builder()
+                .workDir(workingDir)
+                .projectSettings(projectSettings)
+                .environmentVariables(env)
+                .build())) {
+
+            var stackName = randomStackName();
+
+            try {
+                var stack = WorkspaceStack.create(stackName, workspace);
+
+                var upResult = stack.up();
+                assertThat(upResult.summary().kind()).isEqualTo(UpdateKind.UPDATE);
+                assertThat(upResult.summary().result()).isEqualTo(UpdateState.SUCCEEDED);
+                assertThat(upResult.outputs()).hasSize(3);
+
+                var deployment = workspace.exportStack(stackName);
+                assertThat(deployment.version()).isGreaterThan(0);
+
+                var previewBeforeDestroy = stack.preview();
+                assertThat(previewBeforeDestroy.changeSummary().get(OperationType.SAME)).isEqualTo(1);
+
+                stack.destroy();
+
+                var previewAfterDestroy = stack.preview();
+                assertThat(previewAfterDestroy.changeSummary().get(OperationType.CREATE)).isEqualTo(1);
+
+                workspace.importStack(stackName, deployment);
+
+                // After we imported before-destroy deployment,
+                // preview is back to reporting the before-destroy
+                // state.
```

Comment:
This comment isn't super clear to me -- am I missing something? If not, could we make it clearer/remove it if we feel it's not that additive?